tests/integration_tests/tests/agentless_tests/test_labels.py

```diff
@@ -0,0 +1,81 @@
+#########
+# Copyright (c) 2020 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+import uuid
+import pytest
+
+from cloudify.models_states import VisibilityState
+
+from integration_tests.tests import utils
+from integration_tests import AgentlessTestCase
+
+
+@pytest.mark.usefixtures('cloudmock_plugin')
+class DeploymentsLabelsTest(AgentlessTestCase):
+
+    LABELS = [{'key1': 'val1'}, {'key1': 'val2'}, {'key2': 'val3'}]
+    LABELS_2 = [{'key1': 'val1'}, {'key1': 'val3'}, {'key3': 'val3'}]
+
+    def test_viewer_labels_permissions(self):
+        """A "viewer" is allowed to list the labels' keys and values."""
+        self._put_deployment_with_labels(self.client, self.LABELS)
+        self.client.users.create('user', 'password', role='default')
+        self.client.tenants.add_user('user', 'default_tenant', role='viewer')
+        viewer_client = utils.create_rest_client(host=self.env.container_ip,
+                                                 username='user',
+                                                 password='password',
+                                                 tenant='default_tenant')
+
+        keys_list = viewer_client.deployments_labels.list_keys()
+        self.assertEqual(set(keys_list.items), {'key1', 'key2'})
+
+        values_list = viewer_client.deployments_labels.list_key_values('key1')
+        self.assertEqual(set(values_list.items), {'val1', 'val2'})
```

Comment:
You should probably be asserting on all expected labels, just in case we put in a really silly bug in the future. Given that I'm guessing you're doing this multiple times maybe factor out said assertions?

Reply:
Great comment, changing.